fix: Created documentation for /races endpoint

MattMaginniss opened the pull request:
## Why are you making this change?

To start creating a cohesive basic set of documentation for jolpica.

## Changes:

- Created /docs folder.

- Created a basic README.md inside the /docs as the index of the documentation with a list of endpoints, links to their documentation, and a start of a list of known differences between Jolpica and ergast.

Related Issue: https://github.com/jolpica/jolpica-f1/issues/34

## Contributing Checklist
- [X] Unit tests for the changes are included in this PR.
- [X] I have read and agreed to the [contributing guidelines](CONTRIBUTING.md).

jolpica commented on docs/races.md:
```diff
@@ -0,0 +1,204 @@
+
+## Query Parameters:
+
+`limit` - Allows you to alter the limit the number of results returned. Defaults to 30.
+
+`offset` - Allows you to offset the results by the specified number. Defaults to 0.
```
Could we move these into a common markdown file that we can link to from each endpoint?
Would help reduce duplication, and make it easier to update.

MattMaginniss replied:
Makes total sense.

jolpica commented on docs/races.md:
```diff
@@ -0,0 +1,204 @@
+
+**Response Fields** :
+
+`MRData` : The root object of the json response.
+
+`MRData.series` : The racing series that the races are a part of.
+
+`MRData.total` : The total number of races available.
```
Could something similar be done here also? series, total, limit, offset, url, etc all mean the same thing so moving it to a common file which we can link to would be great.

jolpica commented on docs/races.md:
```diff
@@ -0,0 +1,204 @@
+
+## Query Parameters:
+
+`limit` - Allows you to alter the limit the number of results returned. Defaults to 30.
```
Maximum value is 100

jolpica commented on docs/races.md:
```diff
@@ -0,0 +1,204 @@
+`MRData.RaceTable` : The object containing the list of the all races.
+
+`MRData.RaceTable.Races` : The list of all races returned.
+
```
Would be great if all the possible keys inside a race object could be defined, whether they are conditional or not. 
No worries if you're unsure of the details, we can iterate on the docs as we spot things.

MattMaginniss replied:
I'll give it a go

jolpica reviewed (commented):
Thank you for the contribution!
Overall this looks great, just left a few comments on some minor details, and about reducing duplication as we add documentation for more endpoints.

jolpica commented:
Also, we're currently communicating on this slack workspace. Please feel free to join as it may make communication a bit easier.
https://join.slack.com/t/fastf1/shared_invite/zt-2r0nhbegh-FF0y_316FmjBoKh8ou8dSQ

MattMaginniss commented:
Hey @jolpica Added a few things:

Moved common stuff to the index.

<img width="776" alt="image" src="https://github.com/user-attachments/assets/7d50d808-175d-4a32-84a7-4ed4d964d70d">

and added race object fields. did my best to get them all.

<img width="328" alt="image" src="https://github.com/user-attachments/assets/6096bb48-c680-4476-b109-a1bdb3c09850">

MattMaginniss commented:
Hey @jolpica, FYI: For now I will stop with this PR (going to bed) 

If you like the style with the updated structure and decide to merge these, I can create a new branch and PR for the next set of docs.

jolpica reviewed (approved):
Thanks, this is a great start to the docs!
